src/components/twilio-voice-dialer/public/twilio-voice-dialer.js

```diff
@@ -11,12 +11,35 @@ class TwilioVoiceDialer extends HTMLElement {
     this.attachShadow({ mode: 'open' });
   }
 
+  attributeChangedCallback() {
+    this.#render();
+  }
+
+  connectedCallback() {
+    this.shadowRoot
+      .querySelector('#call')
+      .addEventListener('click', () => this.#handleCall());
+    this.shadowRoot
+      .querySelector('#hangup')
+      .addEventListener('click', () => this.#handleHangup());
+  }
+
   disconnectedCallback() {
     this.#device.destroy();
   }
 
-  attributeChangedCallback() {
-    this.#render();
+  async #handleCall() {
+    const recipient = this.shadowRoot.querySelector('#recipient').value;
+    if (recipient) {
+      const call = await this.#device.connect({ params: { recipient } });
+      this.#setupCallHandlers(call);
+      this.#setStatus('inprogress');
+    }
+  }
+
+  #handleHangup() {
+    this.#setStatus('idle');
+    this.#device.disconnectAll();
```

Comment:
Instead of `disconnectAll`, let's save a reference to the `call` object and invoke`call.disconnect()` here.
I understand we're using `disconnectAll()` in our test app. We don't want to do that here :)